### PR TITLE
[REG-1762] Fix error message showing paths not hashes

### PR DESCRIPTION
**Pull request by RG-nAmKcAz**
- Fixes a bug where we showed the hash id instead of the pathname in keyframe errors

---

Find the pull request instructions [here](https://www.notion.so/regressiongg/Pull-Request-Process-0d57a7eb582a446983edfacafa406f1e?pvs=4)

Every reviewer and the owner of the PR should consider these points in their request (feel free to copy this checklist so you can fill it out yourself in the overall PR comment)

- [ ] The code is extensible and backward compatible
- [ ] New public interfaces are extensible and open to backward compatibility in the future
- [ ] If preparing to remove a field in the future (i.e. this PR removes an argument), the argument stays but is no longer functional, and attaches a deprecation warning. A linear task is also created to track this deletion task.
- [ ] Non-critical or potentially modifiable arguments are optional
- [ ] Breaking changes and the approach to handling them have been verified with the team (in the Linear task, design doc, or PR itself)
- [ ] The code is easy to read
- [ ] Unit tests are added for expected and edge cases
- [ ] Integration tests are added for expected and edge cases
- [ ] Functions and classes are documented
- [ ] Migrations for both up and down operations are completed
- [ ] A documentation PR is created and being reviewed for anything in this PR that requires knowledge to use
- [ ] Implications on other dependent code (i.e. sample games and sample bots) is considered, mentioned, and properly handled
- [ ] Style changes and other non-blocking changes are marked as non-blocking from reviewers

### CODE REVIEW

**Review by svolokh — APPROVED** [review]
Non-blocking: I think it would be good to have a more descriptive name than `theVal` for the changes in question. I had to dig deep into the code to figure out what `theVal.Item1` is supposed to represent, which made it somewhat challenging to validate this PR.

Also, I had a hard time making sense of this code from ReplayDataContainer. Is there a bug here? Why is the else branch still reading from `val` if `TryGetValue` returned false? https://github.com/Regression-Games/RGUnityBots/blob/7b7d380dc27133527727e04726493e2965ca9cec/src/gg.regression.unity.bots/Runtime/Scripts/StateRecorder/ReplayDataContainer.cs#L273-L280

Why is it reading

**Comment by RG-nAmKcAz** [comment]
@svolokh please don't approve the PR if you find valid bugs like that, use request changes on those lines

what you reported is a real bug, i'll look into a fix, but please update this to request changes

**Comment by svolokh** [comment]
@RG-nAmKcAz noted! The lines I noted actually were not technically part of this PR's changes, so I don't have a way to directly reference them from within the "Files Changed" tab. I will change my review to request changes.

**Comment by RG-nAmKcAz** [comment]
@svolokh Good catch, should be resolved now

**Review by svolokh — APPROVED** [review]

**Review by vontell — APPROVED** [review]

**Review by abeizer — APPROVED** [review]

**Pull request merged**
